### PR TITLE
researcher: Change beacon deferment fix to reference nActiveBeforeSB

### PR DESCRIPTION
**Pull request by jamescowens**

### CODE REVIEW

**Review by cyrossignol — APPROVED** [review]

File: src/gridcoin/researcher.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "gridcoin/researcher.h"
 #include "gridcoin/support/xml.h"
 #include "gridcoin/tally.h"
+#include "gridcoin/scraper/scraper.h"
```
Comment (cyrossignol) [inline src/gridcoin/researcher.cpp]: `nActiveBeforeSB` is defined in the header. By including it here, the value won't change if the protocol entry changes. To do it that way, the header should declare the variable instead (or use a standalone declaration):

```c++
extern unsigned int nActiveBeforeSB;
```

**Review by cyrossignol — CHANGES_REQUESTED** [review]

File: src/gridcoin/researcher.cpp (hunk 1 above)
Comment (cyrossignol) [reply src/gridcoin/researcher.cpp]: Not entirely sure why the linker didn't complain about the ODR.

Comment (jamescowens) [reply src/gridcoin/researcher.cpp]: You're right of course. Brain dead today.

Comment (jamescowens) [reply src/gridcoin/researcher.cpp]: It wouldn't have changed with your code either... :)

Comment (cyrossignol) [reply src/gridcoin/researcher.cpp]: Heh, probably compiles to the same machine code too :)

**Review by cyrossignol — APPROVED** [review]

File: src/gridcoin/researcher.cpp (hunk 1 above)
Comment (jamescowens) [reply src/gridcoin/researcher.cpp]: Yep. :)

**Pull request merged**